rotkehlchen/api/v1/encoding.py

```diff
@@ -776,6 +776,18 @@ class TradeSchema(Schema):
     link = fields.String(missing='')
     notes = fields.String(missing='')
 
+    @validates_schema  # type: ignore
```

Comment:
As I mentioned in the standup this is not how I wanted this implemented.

If we wanted the price to be zero in some places and not zero in some others then you should add an `__init__` to the `PriceField` with a proper argument, signifying if the particular price field is allowed to be zero or not. Similar to what we do with the asset field here: https://github.com/rotki/rotki/blob/73366f562a95069355f8ab8bcacd67c07ecbe705/rotkehlchen/api/v1/encoding.py#L342-L344

But since `PriceField` is only used at that one place you can get away with just adding an:

```python
if price == ZERO:
    raise ValidationError('A zero rate is not allowed')
```

Notice that since it happens inside the validation of `Price` field you don't need to input the field_name. It will be included in the response.